recipe/meta.yaml

```
@@ -132,7 +132,8 @@ outputs:
         - zstd
       run_constrained:
         - apache-arrow-proc =*={{ build_ext }}
-        - cudatoolkit >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        - cuda-version >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        - cudatoolkit >={{ cuda_compiler_version_min }}   # [cuda_compiler_version != "None"]
```

I guess this is the time to properly tackle #942. My last comment there was:

> [...] it seems to me from re-reading the discussion that we can completely side-step `cuda-version` here, and just run-depend on `__cuda >=9` for the CUDA builds? WDYT @kkraus14

Keith reacted with 👍 there, so if you can live with this as well, we could avoid restarting/rehashing that discussion. 🙃 

Otherwise, let's continue the discussion it in that issue, which has a lot of the background already.

```suggestion
        - __cuda >= 9   # [cuda_compiler_version != "None"]
```

---

+1 to only constraining this to the driver version, the proposed change here LGTM

---

This is in `run_constrained` (not `run`). So am a little confused on the issue this presents (or how it relates to the linked issue discussing `run`). Could someone please clarify?

---

That if someone wants the CUDA enabled build it should only require a CUDA driver and doesn't actually require any of the runtime libraries.

---

Think we are saying the same thing. `run_constrained` doesn't install any of those libraries (only constrains the versions of them to be compatible if the user installs them)

---

I don't think there's any specific issue here other than it being logically incorrect. The Arrow package doesn't use the `cudatoolkit` or anything that is versioned according to `cuda-version` and only uses the cuda driver API which is represented by the `__cuda` package.

---

Sorry, the confusion was mostly my fault in how I proposed the diff. I had written:

> we can completely side-step `cuda-version` here, and just run-depend on `__cuda >=9` for the CUDA builds?

So I was talking about `run:`, not `run_constrained:`

---

Alright am ok with Axel's proposed change. Will go ahead and commit it. Can follow up on anything else that comes up after

---

> > So I was talking about `run:`, not `run_constrained:`
>
> Alright am ok with Axel's proposed change.

Are we sure we're still talking about the same thing? It should go under `run:`, it's now under `run_constrained`.

---

Ok I don't think I understood this to be let's move it from `run_constrained` to `run`

In any event let's discuss this in the new thread where this ask has been clarified ( https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079#discussion_r1222345851 )